test/integration/detachable_joint.cc

```diff
@@ -214,6 +216,77 @@ TEST_F(DetachableJointTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(LinksInSameModel))
   EXPECT_GT(b2Poses.front().Pos().Z() - b2Poses.back().Pos().Z(), expDist);
 }
 
+/////////////////////////////////////////////////
+TEST_F(DetachableJointTest, NestedModelsWithSameName)
+{
+  using namespace std::chrono_literals;
+
+  this->StartServer(common::joinPaths("/test", "worlds",
+       "detachable_joint_child.sdf"));
+
+
+  std::vector<math::Pose3d> childM4Poses, childM5Poses;
+  test::Relay testSystem1;
+  testSystem1.OnPostUpdate([&childM4Poses, &childM5Poses](
+    const gazebo::UpdateInfo &,
+    const gazebo::EntityComponentManager &_ecm)
+  {
+    auto childModels = _ecm.EntitiesByComponents(
+      components::Model(), components::Name("child_model"));
+
+    auto entityM5 = _ecm.EntityByComponents(
+      components::Model(), components::Name("M5"));
+
+    Model modelM5(entityM5);
+    auto childModelsM5 = modelM5.Models(_ecm);
+
+    Entity childEntityM5{kNullEntity}, childEntityM4{kNullEntity};
+    for(auto entity : childModelsM5)
+    {
+      if (entity == childModels[0])
+      {
+        childEntityM5 = childModels[0];
+        childEntityM4 = childModels[1];
+      }
+      if (entity == childModels[1])
+      {
+        childEntityM5 = childModels[1];
+        childEntityM4 = childModels[0];
+      }
+    }
+
+    Model childModelM4(childEntityM4);
+    Model childModelM5(childEntityM5);
+
+    auto poseM4 = _ecm.Component<components::Pose>(childEntityM4);
+    auto poseM5 = _ecm.Component<components::Pose>(childEntityM5);
+
+    childM4Poses.push_back(poseM4->Data());
+    childM5Poses.push_back(poseM5->Data());
+  }
+  );
+
+  this->server->AddSystem(testSystem1.systemPtr);
+
+  const std::size_t nIters{20};
+  this->server->Run(true, nIters, false);
+
+  // Both children of model1 and model4 should not move as they are held
+  // in place
+  EXPECT_EQ(childM4Poses.front(), childM4Poses.back());
+  EXPECT_EQ(childM5Poses.front(), childM5Poses.back());
+
+  // Release M5's child only
+  transport::Node node;
+  auto pub = node.Advertise<msgs::Empty>("/model/M5/detachable_joint/detach");
+  pub.Publish(msgs::Empty());
+  std::this_thread::sleep_for(250ms);
+
+  // Release M5's
```

Comment:
nit: remove this comment

Reply:
[19bdc85](https://github.com/gazebosim/gz-sim/pull/1097/commits/19bdc858b0f28bd1403ece5e1bb95d763a8ac65e)